nmostesting/suites/BCP0060101Test.py

```diff
@@ -35,110 +38,454 @@ def __init__(self, apis):
         GenericTest.__init__(self, apis, omit_paths)
         self.node_url = self.apis[NODE_API_KEY]["url"]
         self.connection_url = self.apis[CONN_API_KEY]["url"]
+        self.is04_resources = {"senders": [], "receivers": [], "_requested": [], "sources": [], "flows": []}
+
+    # Utility function from IS0502Test
+    def get_is04_resources(self, resource_type):
+        """Retrieve all Senders or Receivers from a Node API, keeping hold of the returned objects"""
+        assert resource_type in ["senders", "receivers", "sources", "flows"]
+
+        # Prevent this being executed twice in one test run
+        if resource_type in self.is04_resources["_requested"]:
+            return True, ""
+
+        valid, resources = self.do_request("GET", self.node_url + resource_type)
+        if not valid:
+            return False, "Node API did not respond as expected: {}".format(resources)
+
+        try:
+            for resource in resources.json():
+                self.is04_resources[resource_type].append(resource)
+            self.is04_resources["_requested"].append(resource_type)
+        except json.JSONDecodeError:
+            return False, "Non-JSON response returned from Node API"
+
+        return True, ""
```

Comment:
Should this not be pulled out into NMOSUtils?

Reply:
Yes, for sure, but it currently populates a member variable, so need to work out how best to refactor that, hence left it for later, so far.

Reply:
Yep, sensible plan.